x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -365,7 +366,12 @@ private void authorizeAction(final RequestInfo requestInfo, final String request
                     })
                 );
             });
-            final AsyncSupplier<ResolvedIndices> resolvedIndicesAsyncSupplier = new CachingAsyncSupplier<>(resolvedIndicesListener ->
+            final AsyncSupplier<ResolvedIndices> resolvedIndicesAsyncSupplier = new CachingAsyncSupplier<>(resolvedIndicesListener -> {
+                if (request instanceof IndicesRequest
+                    && false == indicesAndAliasesResolver.requiresWildcardExpansion((IndicesRequest) request)) {
+                    resolvedIndicesListener.onResponse(
+                        indicesAndAliasesResolver.resolveIndicesAndAliases(action, (IndicesRequest) request));
```

Comment:
Breaking off the original `indicesAndAliasesResolver.resolve` method in order to avoid `authorizedIndicesSupplier.getAsync` makes sense.
Three minor quibbles:
* can we integrate the `request instanceof IndicesRequest` inside the `indicesAndAliasesResolver.requiresWildcardExpansion` check method. I think it would be easier to follow.
* can we overload the original `indicesAndAliasesResolver.resolve` method with the new break off method `indicesAndAliasesResolver.resolveIndicesAndAliases` ? or something like: "resolveWithoutWildcards". This is just a naming preference.

Reply:
The tricky bit is that `IndicesRequest` has no relationship with `TransportRequest`. I think that's why existing `IndicesAndAliasesResolver#resolve` takes a `TransportRequest` while `IndicesAndAliasesResolver#resolveIndicesAndAliases` takes a `IndicesRequest`. They are used to handle general requests and requests that are already `IndicesRequest` (bulk action related requests), respectively.

I updated my changes to model after the existing methods:
* `IndicesAndAliasesResolver#resolveWithoutWildcards` is the counterpart for `resolve` and takes a `TransportRequest`. It returns `null` if the give request actually requires expansion.
* `IndicesAndAliasesResolver#resolveIndicesAndAliasesWithoutWildcards` is the break off and counterpart for existing `resolveIndicesAndAliases`. It takes an `IndicesRequest` so that no casting is required for bulk actions.

As a result, I integrated `request instanceof IndicesRequest` inside `resolveWithoutWildcards` as you suggested. Also removed the usage of `authorizedIndicesSupplier` from `authorizeBulkItems` and related methods. I hope these changes addressed your comment.